File: app/src/main/java/com/lucagiorgetti/surprix/model/Producer.java
package com.lucagiorgetti.surprix.model;

/**
 * Created by Utente on 17/04/2017.
 */

public class Producer {
    private String id = null;
    private String name = null;
    private String product = null;
    private String color = null;
    private int order = 0;

    public Producer(String name, String product, int order, String color) {
        this.id = name + "_" + product;
        this.name = name;
        this.product = product;
        this.order = order;
        this.color = color;
    }

    public Producer(String name, int order, String color) {
        this.id = name;
        this.name = name;
        this.color = color;
        this.order = order;
    }

    public Producer() {

    }

    public String getName() {
        return name;
    }

    public String getProduct() {
        return product;
    }

    public String getId() {
        return id;
    }

    public String getColor() {
        return color;
    }

    public int getOrder() {
        return order;
    }

    public void setOrder(int order) {
        this.order = order;
    }
}

File: app/src/main/java/com/lucagiorgetti/surprix/model/Producer.kt
package com.lucagiorgetti.surprix.model

/**
 * Created by Utente on 17/04/2017.
 */
class Producer {
    var id: String? = null
        private set
    var name: String? = null
        private set
    private var product: String? = null
    var color: String? = null
        private set
    var brandId: String? = null
        private set
    var order = 0

    constructor(name: String, product: String?, order: Int, color: String?) {
        id = if (product != null) name + "_" + product else name
        this.name = if (product != null) "$name $product" else name
        brandId = name
        this.product = product
        this.order = order
        this.color = color
    }

    constructor(name: String?, order: Int, color: String?) {
        id = name
        this.name = name
        brandId = name
        this.color = color
        this.order = order
    }

    constructor()
}